### PR TITLE
8. PATCH /api/reviews/:review_id

### PR DESCRIPTION
**Pull request by dafyddhenke**
Card: https://trello.com/c/w02LLAxH. 

- patching review with new votes 
- can increment and decrement votes
- error catches for invalid inc_votes value (such as a string)
- error catches for missing inc_votes key

### CODE REVIEW

**Comment by dafyddhenke** [comment]
Adding a check for when review_id gives no results.

File: __tests__/reviews.controller.test.js [hunk 1]
```diff
@@ -117,3 +120,145 @@ describe("GET /api/reviews/:review_id", () => {
...
+  });
+  test("PATCH /api/reviews/:review_id should output appropriate error messages when passed object without inc_votes key", () => {
+    emptyObject = {
+    };
+    return request(app)
+      .patch("/api/reviews/1")
+      .send(emptyObject)
+      .expect(400)
+      .then((returnedError) => {
+        expect(returnedError.body).toEqual({ msg: "inc_votes is required and value must be greater or less than 0" });
```
Comment (cjplatten) [inline __tests__/reviews.controller.test.js:237]: it is good to test for this! it is worth pointing out that having this test error out could lead to a brittle test in the future if you ever wanted to add another patch functionality. it is up to you if you modify this.

File: __tests__/reviews.controller.test.js [hunk 2]
```diff
@@ -117,3 +120,145 @@ describe("GET /api/reviews/:review_id", () => {
...
+      })
+  });
+  test("PATCH /api/reviews/:review_id should output appropriate error messages when passed inc votes with 0 value", () => {
+    invalidVotes = {
+      inc_votes: 0
+    };
+    return request(app)
+      .patch("/api/reviews/1")
+      .send(invalidVotes)
+      .expect(400)
```
Comment (cjplatten) [inline __tests__/reviews.controller.test.js:259]: would this be an error? if you pass an increment of 0 and the db request increments the votes by 0 that is still a successful request.

File: __tests__/reviews.controller.test.js [hunk 3]
```diff
@@ -117,3 +120,145 @@ describe("GET /api/reviews/:review_id", () => {
...
+      .then((returnedError) => {
+        expect(returnedError.body).toEqual({ msg: "inc_votes is required and value must be greater or less than 0" });
+      })
+  });
+});
```
Comment (cjplatten) [inline __tests__/reviews.controller.test.js:264]: what if the review id doesn't exist or is invalid?

**Review by cjplatten — COMMENTED** [review]
good work so far, please see comments

File: __tests__/reviews.controller.test.js (hunk 2 above)
Comment (dafyddhenke) [reply __tests__/reviews.controller.test.js:259]: I went backwards and forward on this. 

I think my logic was that the "correct" input was either a number greater than 0 or less than 0, and that it wouldn't be possible to adjust a vote with *anything* else. 

I can change it if oyu feel strongly about it.

File: __tests__/reviews.controller.test.js (hunk 3 above)
Comment (dafyddhenke) [reply __tests__/reviews.controller.test.js:264]: I don't know if you saw "Adding a check for when review_id gives no results.". Annoyingly, GitHub is being funky for whatever reason. Now that it seems to be behaving, I'll try and push this update again.

**Pull request merged**